virtualmachine/vsphere/util.go

```diff
@@ -1390,10 +1417,10 @@ var createTemplateName = func(t string, ds string) string {
 var uploadTemplate = func(vm *VM, dcMo *mo.Datacenter, selectedDatastore string) error {
 	var template string
 	if vm.UseLocalTemplates {
-		template = createTemplateName(vm.Template, selectedDatastore)
-	} else {
-		template = vm.Template
+		template = createTemplateName(vm.Template.Name, selectedDatastore)
```

Comment:
club these lines if we r not using template anywhere else.

Reply:
We are using it below in same function.